charts/images.yaml

```diff
@@ -58,10 +58,21 @@ images:
 - name: kube-proxy
   sourceRepository: github.com/kubernetes/kubernetes
   repository: k8s.gcr.io/kube-proxy
+- name: cluster-autoscaler
+  sourceRepository: github.com/gardener/autoscaler
+  repository: eu.gcr.io/gardener-project/gardener/autoscaler/cluster-autoscaler
+  tag: "v1.21.0"
+  targetVersion: "1.21.x"
```

Comment:
What about `> 1.21`? Which version will they use?

Reply:
As per https://github.com/Masterminds/semver#hyphen-range-comparisons 1.21.x will match for range [1.21.0, 1.22.0) which also matches the criteria of using the same major and minor version of k8s version and CA version.
For every new minor version a new entry will have to be added in this yaml file.

Reply:
From what I see for 1.22 and 1.23 the imagevector pkg will fail that no image is provided.
This is also why e2e tests fail:
```
{"level":"info","msg":"At 2022-05-11 12:14:50 +0000 UTC - event for wake-up-p7twj: {gardenlet } ReconcileError: Create botanist failed (retry failed with context deadline exceeded, last error: could not find image \"cluster-autoscaler\" opts runtime version v1.21.1 target version 1.23.6)","ts":"2022-05-11T12:19:49.978Z"}
```

Reply:
added in [5e905c0](https://github.com/gardener/gardener/pull/5955/commits/5e905c09ad0ddba660ae0191614c15a2b6157dbd)

Reply:
for now since there is not release > 1.21 for autoscaler, so we should use 1.21.0 , until later versions are released.